src/randomWallpaperMenu.ts

```diff
@@ -251,36 +264,38 @@ class RandomWallpaperMenu {
         /**
          * Function for events that should happen on element leave.
          *
-         * @param {CustomElements.HistoryElement} unusedActor The activating panel item
-         * @this RandomWallpaperMenu
+         * @param {HistoryEntry} _entry The left/unfocused history entry
          */
-        function onLeave(this: RandomWallpaperMenu, unusedActor: CustomElements.HistoryElement): void {
+        const onLeave = (_entry: HistoryEntry): void => {
+            if (this._savedBackgroundUri)
+                this.previewWidget.preview(GLib.filename_from_uri(this._savedBackgroundUri)[0]);
```

Comment:
`this._savedBackgroundUri` is the current background at menu opening. This doesn't have to align with our "current" history entry when the user changed the background manually without this extension.

You probably want to use this extensions current history element here to get the URI and don't use `this._savedBackgroundUri`.

FYI: The use of `this._savedBackgroundUri` two lines below is correct so the actual live preview of the background gets changed back to the users background even when it's changed manually.

Reply:
Thanks for looking into these changes!
Originally, I thought it was a good idea to have the preview widget show what the current wallpaper actually is. So this wasn't exactly a bug, but I now think it might not be the best idea.

I am testing removing the "Current Background" element entirely to avoid confusion/inconsistency when the wallpaper is changed by the user without the extension. The preview would then only show images from the history (the first element in the history by default). 

These changes are pretty much work in progress, so I'll also change the PR to a draft. But feel free to give it a try if you want. 
![screenshot](https://github.com/ifl0w/RandomWallpaperGnome3/assets/5549805/d6f34d14-06d6-439c-8d28-fce49f722bbf)